### PR TITLE
loosen remaining core dependency versions

### PR DESCRIPTION
**Pull request by natestemen**
## Description

In https://github.com/unitaryfund/mitiq/pull/1672 we loosened the version requirement for `numpy` and haven't had any complaints from users in regards to irreproducible bugs/inconsistencies.

We could do each one of these dependencies at a time, but considering how little we use `scipy`, I think this should be fine to go out as one. Happy to change if others have different opinions.

Fixes https://github.com/unitaryfund/mitiq/issues/1667

### CODE REVIEW

**Review by andreamari — APPROVED** [review]
LGTM! 
Thanks for catching this old issue. I almost forgot about it.

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 numpy>=1.22.0
-scipy~=1.10.1
-cirq~=1.1.0
+scipy>=1.5.0,<=1.10.0
```
Comment (andreamari) [inline requirements.txt:2]: What is the reason for the upper bound `<=1.10.0` for `scipy`? And why is it smaller than `1.10.1` (current setting)?

**Review by andreamari — APPROVED** [review]
Thanks for catching this old issue!
See minor comment about the scipy version upper bound.
After clarifying this small aspect, LGTM!

File: requirements.txt (hunk 1 above)
Comment (natestemen) [reply requirements.txt]: Good point. I think I was just testing stuff and used 1.10 for simplicity. I've bumped to the lastest working version for us in [`46b0237`](https://github.com/unitaryfund/mitiq/pull/1917/commits/46b02375640c4ed47b4a791525a49eed52c2532e).

**Pull request merged**